### PR TITLE
mergeMap new format update

### PR DESCRIPTION
**Pull request by yuHLe**
Added the Why Use section for further explanation of the operator and a memory leak example to demonstrating the danger of misusing the operator.

### CODE REVIEW

File: operators/transformation/mergemap.md [hunk 1]
```diff
@@ -12,6 +12,20 @@
 
 ---
 
+### Why use `mergeMap`?
+
+`mergeMap` is for you when you want to observe all action from the source's emission
```
Comment (btroncone) [inline operators/transformation/mergemap.md]: This is too technical I think, when would you use it? For http requests? What type? Why would you use this over switchMap?

File: operators/transformation/mergemap.md [hunk 2]
```diff
@@ -12,6 +12,20 @@
...
+after you've mapped them to a new observable. In other word, you want to see what
+each item is doing once you've passed them through your function.
+
+These emission could behave differently depending on the function you provided and 
+herein lies the strength of `mergeMap`.  This operators subcribes to all mapped 
+emissions, giving you new output as they come.  It doesn't wait for anything to
+complete or unsubscribes from anything else.  You simply get an unfiltered stream
+of output.  Keep in mind this is a double edge sword.  If the source or inner
```
Comment (btroncone) [inline operators/transformation/mergemap.md]: I would reword this a bit, if the inner observables are long lived and are not properly cleaned up then you could run into a memory leak. If this is the case, and you still want to use mergeMap you may want to utilize takeUntil or another limiting operator to clean up the subscription at the correct time.

File: operators/transformation/mergemap.md [hunk 3]
```diff
@@ -92,6 +106,23 @@ const example = source.mergeMap(
...
+##### Example 5: mergeMap memory leak
+
+( [jsFiddle](https://jsfiddle.net/ElHuy/31y8cv1h/) )
+
+```js
+//Non-stopping source
+const source = Rx.Observable.interval(100);
+
+//Dangerous remapping
+const example = source.mergeMap(val => Rx.Observable.interval(50));
```
Comment (btroncone) [inline operators/transformation/mergemap.md]: I think this is good but can we think of a more realistic scenario? Maybe something from clicks that map to an inner observable? An example of it not getting cleaned up correctly + how to fix?

Comment (yuHLe) [reply operators/transformation/mergemap.md]: I think I have an idea in mind involving user interaction.

File: operators/transformation/mergemap.md (hunk 2 above)
Comment (yuHLe) [reply operators/transformation/mergemap.md]: Probably should mention concurrency here.

**Comment by btroncone** [comment]
lgtm 👍
